Aro doesn't define `_CALL_ELF` correctly for PowerPC

```
docs
└─ langref
   └─ install generated to doc/langref.html
      └─ run exe docgen (langref.html)
         └─ WriteFile test_while_continue_expression.out
            └─ run exe doctest (verbose_cimport_flag.out) failure
info(compilation): writing C import source to /home/ci/zig/zig-local-cache/tmp/1fb593d3c69afaf6/cimport.h
--zig-integration -x c -isystem /home/ci/zig/lib/compiler/aro/include -fno-PIE -fPIC -gdwarf-4 -gdwarf32 -fno-lto -MD -MV -MF /home/ci/zig/zig-local-cache/tmp/1fb593d3c69afaf6/cimport.h.d -fhosted -nostdinc -D__GLIBC_MINOR__=41 -isystem /home/ci/zig/lib/include -isystem /usr/include -isystem /usr/include/powerpc64le-linux-gnu -O0 -target powerpc64le-linux.6.12.57...6.12.57-gnu.2.41 -mcpu=pwr10+64bit-64bitregs+allow_unaligned_fp_access+altivec-booke+bpermd+cmpb+crbits+crypto+direct_move-e500-efpu2+extdiv+fast_MFLR+fcpsgn-float128+fpcvt+fprnd+fpu+fre+fres+frsqrte+frsqrtes+fsqrt+fuse_add_logical-fuse_addi_load-fuse_addis_load+fuse_arith_add-fuse_back2back-fuse_cmp+fuse_logical+fuse_logical_add+fuse_sha3+fuse_store-fuse_wideimm-fuse_zeromove+fusion+hard_float-htm+icbt-invariant_function_descriptors-isa_future_instructions+isa_v206_instructions+isa_v207_instructions+isa_v30_instructions+isa_v31_instructions+isel+ldbrx+lfiwax-longcall+mfocrf+mma-msync+paired_vector_memops+partword_atomics+pcrelative_memops+popcntd+power10_vector+power8_altivec+power8_vector+power9_altivec+power9_vector-ppc4xx-ppc6xx+ppc_postra_sched+ppc_prera_sched+predictable_select_expensive+prefix_instrs-privileged+quadword_atomics+recipprec-rop_protect-secure_plt-slow_popcntd-spe+stfiwx+two_const_nr-vectors_use_two_units+vsx /home/ci/zig/zig-local-cache/tmp/1fb593d3c69afaf6/cimport.h -o /home/ci/zig/zig-local-cache/tmp/1fb593d3c69afaf6/cimport.zig
info(compilation): processing dep file at /home/ci/zig/zig-local-cache/tmp/1fb593d3c69afaf6/cimport.h.d
/home/ci/zig/doc/langref/verbose_cimport_flag.zig:1:11: error: C import failed
const c = @cImport({
          ^~~~~~~~
referenced by:
    main: /home/ci/zig/doc/langref/verbose_cimport_flag.zig:6:9
    callMain [inlined]: /home/ci/zig/lib/std/start.zig:705:22
    callMainWithArgs [inlined]: /home/ci/zig/lib/std/start.zig:674:20
    main: /home/ci/zig/lib/std/start.zig:689:28
    1 reference(s) hidden; use '-freference-trace=5' to see all references
error: translation failure
/usr/include/powerpc64le-linux-gnu/gnu/stubs.h:11:11: error: 'gnu/stubs-64-v1.h' not found
# include <gnu/stubs-64-v1.h>
          ^
```

It should be taking the `stubs-64-v2.h` path.

--- a/src/aro/Compilation.zig
+++ b/src/aro/Compilation.zig
@@ -647,6 +647,7 @@ fn generateSystemDefines(comp: *Compilation, w: *Io.Writer) !void {
             try define(w, "__ppc__");
             try define(w, "__PPC__");
             try define(w, "_ARCH_PPC");
+            try w.print("#define _CALL_ELF {d}\n", .{target.ppcElfVersion()});
         },
         .powerpc64,
         .powerpc64le,
@@ -661,6 +662,7 @@ fn generateSystemDefines(comp: *Compilation, w: *Io.Writer) !void {
             try define(w, "__PPC64__");
             try define(w, "_ARCH_PPC");
             try define(w, "_ARCH_PPC64");
+            try w.print("#define _CALL_ELF {d}\n", .{target.ppcElfVersion()});
         },
         .sparc64 => {
             try defineStd(w, "sparc", is_gnu);

--- a/src/aro/Target.zig
+++ b/src/aro/Target.zig
@@ -1348,6 +1348,18 @@ pub fn isPIEDefault(target: *const Target) DefaultPIStatus {
     };
 }
 
+pub fn ppcElfVersion(target: *const Target) u32 {
+    std.debug.assert(target.cpu.arch.isPowerPC());
+    return switch (target.cpu.arch) {
+        .powerpc64le => 2,
+        .powerpc64 => if ((target.os.isAtLeast(.freebsd, .{ .major = 13, .minor = 0, .patch = 0 }) orelse false) or target.os.tag == .openbsd or target.abi.isMusl())
+            2
+        else
+            1,
+        else => 1,
+    };
+}
+
 pub fn isPICdefault(target: *const Target) DefaultPIStatus {
     return switch (target.os.tag) {
         .haiku,

--- a/test/cases/ppc elf v1.c
+++ b/test/cases/ppc elf v1.c
@@ -1,0 +1,6 @@
+_Static_assert(_CALL_ELF == 1, "Wrong _CALL_ELF value");
+
+/** manifest:
+syntax
+args = --target=powerpc64-linux-gnu
+*/

--- a/test/cases/ppc elf v2.c
+++ b/test/cases/ppc elf v2.c
@@ -1,0 +1,6 @@
+_Static_assert(_CALL_ELF == 2, "Wrong _CALL_ELF value");
+
+/** manifest:
+syntax
+args = --target=powerpc64-freebsd.13.1-gnu
+*/

